middleware/cors.go

```diff
@@ -189,11 +210,13 @@ func CORSWithConfig(config CORSConfig) echo.MiddlewareFunc {
 			// Preflight request
 			res.Header().Add(echo.HeaderVary, echo.HeaderAccessControlRequestMethod)
 			res.Header().Add(echo.HeaderVary, echo.HeaderAccessControlRequestHeaders)
-			res.Header().Set(echo.HeaderAccessControlAllowOrigin, allowOrigin)
-			res.Header().Set(echo.HeaderAccessControlAllowMethods, allowMethods)
-			if config.AllowCredentials {
-				res.Header().Set(echo.HeaderAccessControlAllowCredentials, "true")
+
+			if !hasCustomAllowMethods && routerAllowMethods != "" {
```

Comment:
this IF/ELSE is meant to be as: if CORS knows "most" correct info about which routes are supported (from `echo.ContextKeyHeaderAllow` context key) it will use to as header. If it was not set or somethings has unset it - it will fallback to old way of settings header value from config struct. Also if user has set their own allow methods in config - do not use router knowledge for header value.